src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,202 @@
+        shooterMotor.set(power);
+    }
+
+    /**
+     * Sets the motor to run at a specific voltage.
+     *
+     * @param volts The voltage.
+     */
+    public void setMotorVolts(double volts) {
+        shooterMotor.setVoltage(volts);
+    }
+
+    /**
+     * Sets the motor to run to a target speed in RPM.
+     *
+     * @param speed The target speed in RPM.
+     */
+    public void setMotorTargetSpeed(double speed) {
+        speed = Functions.clampDouble(speed, MAX_RPM, -MAX_RPM);
+        shooterMotorPIDController.setReference(speed, CANSparkMax.ControlType.kVelocity);
+    }
+
+    /**
+     * Sets the shooter encoder value.
+     *
+     * @param position Value to set it to
+     */
+    public void setEncoderValue(double position) {
+        shooterEncoder.setPosition(position);
+    }
+
+    /**
+     * Gets the shooter encoder value.
+     *
+     * @return the encoder value
+     */
+    public double getEncoderValue() {
+        return shooterEncoder.getPosition();
+    }
+
+    /**
+     * Zeros the shooter encoder(s).
+     * Used the plural version of encoder for consistancy throughout the program.
+     */
+    public void zeroEncoders() {
+        setEncoderValue(0);
+    }
+
+    /**
+     * Gets the current shooter position in rotations.
+     *
+     * @return The current shooter position
+     */
+    public double getShooterPosition() {
+        return shooterEncoder.getPosition();
+    }
+
+    /**
+     * Gets the current shooter velocity in RPM.
+     *
+     * @return the shooter velocity in RPM
+     */
+    public double getShooterVelocity() {
+        return shooterEncoder.getVelocity();
+    }
+
+    /**
+     * Sets the ramp rate in a closed environment.
+     * EG. When under PID
+     *
+     * @param rate The rate to set it to.
+     */
+    public void setClosedRampRate(double rate) {
+        shooterMotor.setClosedLoopRampRate(rate);
+    }
+
+    /**
+     * Sets the ramp rate in an open environment.
+     * EG. When setting the motor power
+     *
+     * @param rate The rate to set it to.
+     */
+    public void setOpenRampRate(double rate) {
+        shooterMotor.setOpenLoopRampRate(rate);
+    }
+
+    /**
+     * Gets the current PID values.
+     *
+     * @return the pid values as an array of doubles.
+     */
+    public double[] getPID() {
+        return new double[] {P, I, D};
+    }
+
+    /**
+     * Extends the shooter hood.
+     */
+    public void extendHood() {
+        hoodPos = true;
+        hood.set(DoubleSolenoid.Value.kForward);
+    }
+
+    /**
+     * Retracts the shooter hood.
+     */
+    public void retractHood() {
+        hoodPos = false;
+        hood.set(DoubleSolenoid.Value.kReverse);
+    }
+
+    /**
+     * Gets the current hood position.
+     * True - Piston Extended
+     * False - Piston Not Extended
+     *
+     * @return The current hood position.
+     */
+    public boolean getHoodPos() {
+        return hoodPos;
+    }
+
```

Comment:
a setHoodPos(bool pos) would be nice for binding to a button, make sure it caches as well